textile/features.textile

```diff
@@ -1655,6 +1656,17 @@ h4. DevicePushDetails
 * @(PCP3)@ recipient - a map of string key/value pairs containing details of the push transport and address
 * @(PCP4)@ state - the state of the push registration, one of  @Active@, @Failing@, @Failed@
 
+h3(#introspection). Client library introspection
+
+h4. ClientLibraryInfo
+* @(CLI1)@ Provides information about the client library.
+* @(CLI2)@ @agents@ static property:
+** @(CLI2a)@ Returns a @Dict<String, String?>@ that lists the default key-value entries that the library uses to populate the @Agent@ library identifier, as described by "@RSC7d@":#RSC7d.
+* @(CLI3)@ @agentIdentifier@ static method:
+** @(CLI3a)@ Returns the @Agent@ library identifier described by "@RSC7d@":#RSC7d.
+** @(CLI3b)@ If the library offers the @ClientOptions#agents@ property described by "@RSC7d6@":#RSC7d6, then the @agentIdentifier@ method must accept an @additionalAgents@ parameter of type @Dict<String, String?>?@, whose value is to be interpreted with the same semantics as the @ClientOptions#agents@ property. The @agentIdentifier@ method will inject these agents into the @Agent@ library identifier that it returns.
```

Comment:
`agents` being presented in two places looks like we couldn't decide the right one tbh.

Reply:
Hey @maratal, I'm not sure I understand what you mean here. Would you be able to elaborate a bit please?

Reply:
I mean `ClientLibraryInfo.agents` and 'ClientOptions.agents'

Reply:
They do different things though – `ClientLibraryInfo.agents` is for querying the library’s version, and `ClientOptions.agents` is for telling the library which additional agents to use.